arangod/Utils/CollectionNameResolver.cpp

```diff
@@ -40,6 +40,15 @@ std::string const UNKNOWN("_unknown");
 
 namespace arangodb {
 
+// copy an existing resolver
+CollectionNameResolver::CollectionNameResolver(CollectionNameResolver const& other) 
```

Comment:
why is this copyable?? aren't these thread-safe

Reply:
Previously, the class was default-copyable and copy-assignable (which would not acquire any locks and wouldn't be safe). Now only copy construction is allowed and everything else is forbidden. Copy construction is implemented safely.
Right now we don't copy resolvers, but we may want to do this at some point instead of use one resolver (and sharing it). Or just to prime a new CollectionNameResolver from an existing one etc.